app/src/main/java/com/example/num1/SignInActivity.kt

```diff
@@ -34,7 +33,7 @@ class SignInActivity : AppCompatActivity() {
             { result ->
                 if (result.resultCode == Activity.RESULT_OK) {
                     val data: Intent? = result.data
-                    val userId = data?.getStringExtra("id")
+                    val userId = data?.getStringExtra("id") //signup에서 넣어준걸 get으로 받아준다.
                     val userPw = data?.getStringExtra("pw")
```

Comment:
```suggestion
                    result.data?.let {
                        val userId = it.getStringExtra("id") //signup에서 넣어준걸 get으로 받아준다.
                        val userPw = it.getStringExtra("pw")
                    }
```
`result.data?.let { }`은, result.data가 null이면 중괄호 안의 코드를 수행하지 않습니다. 다시 말하면, let 왼쪽의 변수가 null이 아니어야 중괄호 안의 코드가 실행됩니다. 이렇게 코드를 작성할 때의 좋은 점은 이렇습니다.

> 중괄호 안의 코드를 수행한다는 것은 result.data가 null이 아님을 의미 -> 불필요한 `?`를 계속 쓰지 않아도 됨